### PR TITLE
chore: Change get_table_names/get_view_names return type

### PR DESCRIPTION
**Pull request by john-bodley**
### SUMMARY

After working on https://github.com/apache/superset/pull/21794 I realized that the `get_table_names` and `get_view_names` methods should really return a set of names rather than an ordered list. The results can then be sorted—if needed—within the API response. Hopefully this refactor simplifies the code logic.

### TESTING INSTRUCTIONS

CI.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/models/core.py [hunk 1]
```diff
@@ -556,13 +556,17 @@ def get_all_table_names_in_schema(  # pylint: disable=unused-argument
         :param cache: whether cache is enabled for the function
         :param cache_timeout: timeout in seconds for the cache
         :param force: whether to force refresh the cache
-        :return: list of tables
+        :return: set of tables
```
Comment (villebro) [inline superset/models/core.py:559]: code style: Some people consider including the type in the return description bad practice if it's already typed in the sig. I'm not sure which camp I belong to (it varies from day to day), but maybe we could rather try to be more expressive regarding the funny `Tuple[str, str]` thingy, something like
```
:return: table name and schema pairs
```

File: tests/integration_tests/db_engine_specs/base_engine_spec_tests.py [hunk 2]
```diff
@@ -229,11 +229,11 @@ def test_get_table_names(self):
 
         """ Make sure base engine spec removes schema name from table name
         ie. when try_remove_schema_from_table_name == True. """
-        base_result_expected = ["table", "table_2"]
+        base_result_expected = {"table", "table_2"}
         base_result = BaseEngineSpec.get_table_names(
             database=mock.ANY, schema="schema", inspector=inspector
         )
-        self.assertListEqual(base_result_expected, base_result)
+        self.assertSetEqual(base_result_expected, base_result)
```
Comment (villebro) [inline tests/integration_tests/db_engine_specs/base_engine_spec_tests.py:236]: could we just do this as that's what appears to be preferred these days?
```
assert base_result_expected == base_result
```

**Review by villebro — APPROVED** [review]
LGTM with some besserwisser-comments

**Comment by ktmud** [comment]
I don't understand why you have to return set. Shouldn't DBAPI return unique table names to begin with?

File: tests/integration_tests/db_engine_specs/base_engine_spec_tests.py (hunk 2 above)
Comment (john-bodley) [reply tests/integration_tests/db_engine_specs/base_engine_spec_tests.py:236]: I agree. I didn't know if unittest supported this.

**Comment by ktmud** [comment]
Set has the additional memory cost of keeping the hashtable of elements: https://towardsdatascience.com/memory-efficiency-of-common-python-data-structures-88f0f720421

Maybe not a big deal in this case but in general I don't think we should sacrifice performance for an implied convention that is supposed to help with readability.

Regarding readability, in the case where sets are used, people would wonder "why this HAS to be a set" (like I did), which actually makes the code more confusing to them.

**Pull request merged**